archunit/src/test/java/com/tngtech/archunit/core/domain/testobjects/ClassWithDependencyOnTryCatchBlock.java

```diff
@@ -0,0 +1,39 @@
+package com.tngtech.archunit.core.domain.testobjects;
+
+import java.io.IOException;
+
+@SuppressWarnings("unused")
+public class ClassWithDependencyOnTryCatchBlock {
+    static {
+        try {
+            throw new IOException();
+        } catch (IOException e) {
+        }
+    }
+
+    ClassWithDependencyOnTryCatchBlock()
+    {
+        try {
+            throw new IOException();
```

Comment:
I'd use different exceptions for each of the cases to make the test even more clear.